reactivesocket/src/main/scala/akka/stream/alpakka/reactivesocket/scaladsl/HelloWorld.scala

```diff
@@ -0,0 +1,163 @@
+package akka.stream.alpakka.reactivesocket.scaladsl
+
+import io.reactivesocket.transport.tcp.server.TcpTransportServer
+import io.reactivesocket.server.ReactiveSocketServer
+import io.reactivesocket.server.ReactiveSocketServer.SocketAcceptor
+import io.reactivesocket.ConnectionSetupPayload
+import io.reactivesocket.lease.LeaseEnforcingSocket
+import io.reactivesocket.ReactiveSocket
+import io.reactivesocket.lease.DisabledLeaseAcceptingSocket
+import io.reactivesocket.AbstractReactiveSocket
+import org.reactivestreams.Publisher
+import io.reactivesocket.Payload
+import io.reactivesocket.client.ReactiveSocketClient
+import io.reactivesocket.transport.tcp.client.TcpTransportClient
+
+import io.reactivesocket.client.KeepAliveProvider._
+import io.reactivesocket.client.SetupProvider._
+import akka.actor.ActorSystem
+import akka.stream.ActorMaterializer
+import akka.stream.scaladsl.Source
+import akka.stream.scaladsl.Sink
+import io.reactivesocket.util.PayloadImpl
+import scala.concurrent.Future
+import io.reactivesocket.frame.ByteBufferUtil
+import akka.stream.Materializer
+import akka.Done
+import akka.NotUsed
+
+object HelloWorld {
+  def main(args: Array[String]): Unit =
+    (new HelloWorld).start()
+}
+
+class HelloWorld {
+  val system = ActorSystem("HelloWorld")
+  implicit val materializer = ActorMaterializer.create(system)
+  import system.dispatcher
+
+  def start(): Unit = {
+
+    val serverHandler = new AbstractReactiveSocketScaladsl {
+      override def requestResponse(payload: Payload): Future[Payload] = {
+        println(s"# got request") // FIXME
+        Future.successful(payload)
+      }
+    }
+    val server = ReactiveSocketServer.create(TcpTransportServer.create())
+      .start(SocketAcceptorAdapter.disabledLease(serverHandler))
+
+    val serverAddress = server.getServerAddress
+    println(s"started server: $serverAddress") // FIXME
+
+    val client: Future[ReactiveSocketScaladsl] =
+      ReactiveSocketClientScaladsl(ReactiveSocketClient.create(TcpTransportClient.create(serverAddress),
+        keepAlive(never()).disableLease()))
+
+    client.foreach { socket =>
+      println(s"# got socket") // FIXME
+      val response: Future[Payload] = socket.requestResponse(new PayloadImpl("Hello"))
+
+      response.onFailure {
+        case e => println(e.getMessage)
+      }
+
+      response.map(_.getData)
+        .map(x => { println("got response"); x }) // FIXME remove
+        .map(ByteBufferUtil.toUtf8String)
+        .map(println)
+        .andThen { case _ => socket.close() }
+    }
+
+  }
+
+}
+
+trait ReactiveSocketScaladsl {
```

Comment:
Scaladsl should of course not be in the names later, just prototyping the api here